thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2924,12 +3060,29 @@ class HashDedupSlaveActivityBase : public CSlaveActivity
                 lastEog = false;
                 if (bucketHandler->addRow(row)) // true if new, i.e. non-duplicate (does not take ownership)
                 {
-                    dataLinkIncrement();
-                    return row.getClear();
+                    if (!keepBest)
+                    {
+                        dataLinkIncrement();
+                        return row.getClear();
+                    }
                 }
             }
             else
             {
+                if (keepBest && !keepBestRowsReadyToReturn)
+                {
+                    bucketHandler->clearCallbacks();
```

Comment:
Added code to test spilling.  It works because even if a bucket or more it handles that in the section following.  (The stream buckets only handles buckets that haven't been spilt).